ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/GhcrIntegrationForm.tsx

```diff
@@ -0,0 +1,326 @@
+import React, { ReactElement } from 'react';
+import { Flex, FlexItem, TextInput, PageSection, Form, Checkbox } from '@patternfly/react-core';
+import * as yup from 'yup';
+import merge from 'lodash/merge';
+
+import { ImageIntegrationBase } from 'services/ImageIntegrationsService';
+
+import usePageState from 'Containers/Integrations/hooks/usePageState';
+import ExternalLink from 'Components/PatternFly/IconText/ExternalLink';
+import FormMessage from 'Components/PatternFly/FormMessage';
+import FormTestButton from 'Components/PatternFly/FormTestButton';
+import FormSaveButton from 'Components/PatternFly/FormSaveButton';
+import FormCancelButton from 'Components/PatternFly/FormCancelButton';
+import IntegrationHelpIcon from './Components/IntegrationHelpIcon';
+import useIntegrationForm from '../useIntegrationForm';
+import { IntegrationFormProps } from '../integrationFormTypes';
+
+import IntegrationFormActions from '../IntegrationFormActions';
+import FormLabelGroup from '../FormLabelGroup';
+
+export type GhcrIntegration = {
+    categories: 'REGISTRY'[];
+    docker: {
+        endpoint: string;
+        username: string;
+        password: string;
+        insecure: boolean;
+    };
+    type: 'ghcr';
+} & ImageIntegrationBase;
+
+export type GhcrIntegrationFormValues = {
+    config: GhcrIntegration;
+    updatePassword: boolean;
+};
+
+export const validationSchema = yup.object().shape({
+    config: yup.object().shape({
+        name: yup.string().trim().required('An integration name is required'),
+        categories: yup
+            .array()
+            .of(yup.string().trim().oneOf(['REGISTRY']))
+            .min(1, 'Must have at least one type selected')
+            .required('A category is required'),
+        docker: yup.object().shape({
+            endpoint: yup.string().trim().required('An endpoint is required'),
+            username: yup.string(),
+            password: yup
+                .string()
+                .test(
+                    'password-test',
+                    'A password is required',
+                    (value, context: yup.TestContext) => {
+                        const requirePasswordField = !!context?.from?.[2]?.value?.updatePassword;
+                        if (!requirePasswordField) {
+                            return true;
+                        }
+                        return !!value?.trim();
+                    }
+                ),
```

Comment:
I suggest removing the criteria that requires a password to be set to improve UX.

The `Save` button is currently disabled when a password is not set.  Any user that deletes the default integration or upgrades an existing Stackrox installation will be unable to create an empty password GHCR integration via the UI (the default integrations are only created on fresh installs when no other integrations exist).

Reply:
done